src/platform/backends/shared/linux/process_factory.cpp

```diff
@@ -90,7 +99,16 @@ std::unique_ptr<mp::Process> mp::ProcessFactory::create_process(std::unique_ptr<
 {
     if (apparmor && !process_spec->apparmor_profile().isNull())
     {
-        return std::make_unique<AppArmoredProcess>(apparmor.value(), std::move(process_spec));
+        try
+        {
+            return std::make_unique<AppArmoredProcess>(apparmor.value(), std::move(process_spec));
+        }
+        catch (const mp::AppArmorException& e)
+        {
+            // TODO: This won't fly in strict mode (#1074), since we'll be confined by snapd
```

Comment:
What's the plan then?

Reply:
We will have to detect the available AppArmor features and amend the profile to work.